### ISSUE TITLE
[Check] Django check should not force null=True for ManyToManyField(blank=True) 

### ISSUE THREAD

**Issue by askoretskiy** [returntocorp/sgrep-rules#743]
**Describe the bug**

Following example gives an error:

```
class Campaign(models.Model):
    tags = models.ManyToManyField(Tag, blank=True)
```

Error:

```
semgrep --error --force-color --config=https://semgrep.live/c/p/django
models.py
severity:error rule:python.django.correctness.nontext-field-must-set-null-true.nontext-field-must-set-null-true: null=True should be set if blank=True is set on non-text fields.
170:    tags = models.ManyToManyField(Tag, blank=True)
```

At the same time, the source code of Django says that:
```python
        if self.has_null_arg: # `'null' in kwargs`
            warnings.append(
                checks.Warning(
                    'null has no effect on ManyToManyField.',
                    obj=self,
                    id='fields.W340',
                )
            )
```

So if one use `tags = models.ManyToManyField(Tag, blank=True, null=True)` -- it would raise Django warning.

**Expected behavior**
Error is skipped for ManyToManyField

**Possible fix**

If I understand it correctly, this should help:

```
pattern-not: $F = django.db.models.ManyToManyField(...)

```

**Comment by askoretskiy** [comment]
If nobody takes it, I would prepare MRs tomorrow.

**Comment by dlukeomalley** [comment]
@askoretskiy Similar to #742, an MR is appreciated. 

@jacobian @minusworld any thoughts to add here as well?

**Comment by askoretskiy** [comment]
I've created one MR for both issues.

**Comment by jacobian** [comment]
Yeah, this is right. `null=True` is meaningless for `ManyToManyField`s. They're weird in that they're not really fields, since they actually work by generating an intermediary table, so `null=True` doesn't make any sense in that context.

**Comment by dlukeomalley** [comment]
Closed by #745
